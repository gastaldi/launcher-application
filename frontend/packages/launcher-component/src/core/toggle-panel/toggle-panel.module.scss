.panel {
  max-height: 0;
  transition: max-height 0.25s ease-in;
  overflow: hidden;
}

.expanded {
  max-height: 1000px;
}
