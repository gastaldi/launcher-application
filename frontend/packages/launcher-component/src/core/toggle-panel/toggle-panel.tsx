import React, { ReactNode, Fragment } from 'react';
import { Button } from '@patternfly/react-core';
import { MinusCircleIcon, PlusCircleIcon } from '@patternfly/react-icons';

import style from './toggle-panel.module.scss';
import { useSessionStorageWithObject } from 'react-use-sessionstorage';

interface TogglePanelProps {
  title: string;
  children: ReactNode;
}

export function TogglePanel(props: TogglePanelProps) {
  const [collapse, setCollapse] = useSessionStorageWithObject(props.title, false);
  return (
    <Fragment>
      <Button
        // @ts-ignore
        component="a"
        variant="link"
        aria-label="Expand panel"
        onClick={() => setCollapse(!collapse)}
      >
        {collapse ? <MinusCircleIcon /> : <PlusCircleIcon />} {props.title}
      </Button>
      <div className={`${style.panel} ${(collapse ? style.expanded : '')}`}>
        {props.children}
      </div>
    </Fragment>
  );
}
