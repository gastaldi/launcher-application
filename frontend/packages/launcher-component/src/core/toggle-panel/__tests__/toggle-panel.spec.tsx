import * as React from 'react';
import { TogglePanel } from '../toggle-panel';
import { cleanup, fireEvent, render } from 'react-testing-library';

afterEach(cleanup);

describe('<TogglePanel />', () => {
  it('renders the TogglePanel correctly', () => {
    const comp = render(
      <TogglePanel title="TogglePanel">
        <p>the panel content</p>
      </TogglePanel>
    );
    expect(comp.asFragment()).toMatchSnapshot();
  });

  it('show the edition panel for the clicked hub', () => {
    const comp = render(
      <TogglePanel title="TogglePanel">
        <p title="content">the panel content</p>
      </TogglePanel>
    );
    const link = comp.getByLabelText('Expand panel');
    fireEvent.click(link);
    expect(comp.getByTitle('content')).toBeDefined();
    expect(comp.asFragment()).toMatchSnapshot();
  });
});
