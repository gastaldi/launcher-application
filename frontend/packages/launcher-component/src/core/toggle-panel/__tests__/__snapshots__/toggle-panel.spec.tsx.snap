// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TogglePanel /> renders the TogglePanel correctly 1`] = `
<DocumentFragment>
  <a
    aria-disabled="false"
    aria-label="Expand panel"
    class="pf-c-button pf-m-link"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="1em"
      role="img"
      style="vertical-align: -0.125em;"
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
        transform=""
      />
    </svg>
     TogglePanel
  </a>
  <div
    class="panel "
  >
    <p>
      the panel content
    </p>
  </div>
</DocumentFragment>
`;

exports[`<TogglePanel /> show the edition panel for the clicked hub 1`] = `
<DocumentFragment>
  <a
    aria-disabled="false"
    aria-label="Expand panel"
    class="pf-c-button pf-m-link"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="1em"
      role="img"
      style="vertical-align: -0.125em;"
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zM124 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H124z"
        transform=""
      />
    </svg>
     TogglePanel
  </a>
  <div
    class="panel expanded"
  >
    <p
      title="content"
    >
      the panel content
    </p>
  </div>
</DocumentFragment>
`;
